Mathlib/Combinatorics/Quiver/ReflQuiver.lean

```diff
@@ -0,0 +1,129 @@
+/-
+Copyright (c) 2024 Mario Carneiro and Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Emily Riehl
+-/
+import Mathlib.Data.Set.Function
+import Mathlib.CategoryTheory.Category.Cat
+
+/-!
+# Reflexive Quivers
+
+This module defines reflexive quivers. A reflexive quiver, or "refl quiver" for short, extends
+a quiver with a specified endoarrow on each term in its type of objects.
+
+We also introduce morphisms between reflexive quivers, called reflexive prefunctors or "refl
+prefunctors" for short.
+-/
```

Comment:
Maybe some note saying that `Category` doesn't extend `ReflQuiver`, and why?

Reply:
As mentioned in the PR description, this is a TODO. It's pending zulip discussion or so, so there is not much in the way of "why" beyond it being a significant refactor which should go in a separate PR.